### PR TITLE
Add developer docs

### PR DESCRIPTION
**Pull request by n-wbrown**
## Description

Add a short documentation page explaining the steps to setup a development environment for pcdswidgets.

## Motivation and Context

Pcdswidgets is slightly more complex to install in a development environment because it requires an environment variable to be set before the widgets can be accessed in designer. This step was not apparent from the existing documentation so I thought it should be added. 

## How Has This Been Tested?

Having followed these steps, my environment works. 

## Where Has This Been Documented?

This is a documentation PR.

### CODE REVIEW

File: docs/source/dev.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+=======================================
+Configuring the Development Environment
+=======================================
+
+#. Create a virtual environment with conda or use a preexisting environment.
+#. Activate the environment. All following steps assume that you are inside this enviornment.
+#. Install pydm, pyqt, and pcdsdevices from the conda-forge channel with ``conda install -c conda-forge pydm pyqt pcdsdevices``
+#. Clone a copy of pcdswidgets from here ``https://github.com/pcdshub/pcdswidgets`` 
+#. Move into the newly cloned pcdswidgets directory and run ``pip install -e .`` to perform a development install of pcdswidgets.
+#. In order to ensure that designer loads the new widgets add the newly cloned pcdswidgets directory to the environment variable ``PYQTDESGNERPATH``
```
Comment (klauer) [inline docs/source/dev.rst:10]: I think it's:
```suggestion
#. In order to ensure that designer loads the new widgets add the newly cloned pcdswidgets directory to the environment variable ``PYQTDESIGNERPATH``
```

A quick copy/paste-able one would be like:
```bash
export PYQTDESIGNERPATH=$PWD:$PYQTDESIGNERPATH
```

**Review by klauer — APPROVED** [review]
One minor fix to the environment variable name, and a suggestion to make it easily copy/paste-able.

Looks good to me 👍 and especially helpful since setting up the Qt designer to work with PyQt-defined widgets can be exceedingly confusing.

File: docs/source/dev.rst (hunk 1 above)
Comment (n-wbrown) [reply docs/source/dev.rst]: Thanks for catching this @klauer! I've incorporated the fix and suggestion into the next commit.

**Review by klauer — APPROVED** [review]
Looks good, @n-wbrown 👍 Let's merge it in.

**Pull request merged**